command/agent/command.go

```diff
@@ -186,6 +187,8 @@ func (c *Command) readConfig() *Config {
 	// Parse the meta flags.
 	metaLength := len(meta)
 	if metaLength != 0 {
+		validKeyRe, _ := regexp.Compile(`^[^.]+(\.[^.]+)*$`)
```

Comment:
My mental regexp parser says this is ok, but let's put it in a package like [`helper`](https://godoc.org/github.com/hashicorp/nomad/helper) as an exported package-level variable for reuse in the other location:

```go
// VariableRegexp matches valid dotted variable names for interpolation. The string must begin with one or more non-dot characters which may be followed by sequences containing a dot followed by a one or more non-dot characters.
var VariableRegexp = regexp.MustCompile(`^[^.]+(\.[^.]+)*$`)
```

(with proper wrapping on the comment)

Reply:
done.